modules/serverless-logic-overview-timeouts.adoc

```diff
@@ -0,0 +1,205 @@
+
+For example, to cancel the workflow after an hour of execution, use the following configuration:
+
+.Example of workflow timeout
+[source,json]
+----
+  {
+  "id": "workflow_timeouts",
+  "version": "1.0",
+  "name": "Workflow Timeouts",
+  "description": "Simple workflow to show the workflowExecTimeout working",
+  "start": "PrintStartMessage",
+  "timeouts": {
+    "workflowExecTimeout": "PT1H"
+  } ...
+}
+----
+
+[id="event-timeout_{context}"]
+== Event timeout
+
+When you define a state in a workflow, you can use the `timeouts` property to configure the maximum time to complete this state. When that time is overdue, the state is considered timed-out, and the engine continues the execution from this state. The execution flow depends on the state type. For example, a transition to a next state might be executed. 
+
+Event-based states can use the sub-property `eventTimeout` to configure the maximum time to wait for an event to arrive. This is the only property that is supported in the current implementation.
+
+The Event state `timeouts` property can be used to specify state specific timeout settings. For the Event state, it can contain the `eventTimeout` property which is defined using the `ISO 8601` data and time format. For example, you can specify `PT15M` to represent 15 minutes or `P2DT3H4M` to represent 2 days, 3 hours, and 4 minutes. The `eventTimeout` values should always be represented as durations and not as specific time intervals.
+
+If the Event state is a workflow starting state, incoming events might trigger workflow instances. In this case, if the `exclusive` property is set to `true`, the `eventTimeout` property is ignored.
+
+If the `exclusive` property is set to false, in this case, the defined `eventTimeout` represents the time between arrival of specified events.
+
+For example, consider the following example:
+
+.Example for the event timeout
+[source,json]
+----
+{
+"states": [
+{
+    "name": "ExampleEventState",
+    "type": "event",
+    "exclusive": false,
+    "timeouts": {
+      "eventTimeout": "PT2M"
+    }
+    "onEvents": [
+        {
+            "eventRefs": [
+                "ExampleEvent1",
+                "ExampleEvent2"
+            ],
+            "actions": [
+              ...
+            ]
+        }
+    ],
+    "end": {
+        "terminate": true
+    }
+}
+]
+}
+----
+
+The first `eventTimeout` starts after any of the referenced events are consumed. If the second event does not occur within the defined eventTimeout, no workflow instance should be created.
+
+Event timeouts supports callback state timeout, switch state timeout, and event state timeout.
+
+The `Callback` state can be used when you must execute an action in general to call an external service, and wait for an asynchronous response in the form of an event.
```

Comment:
```suggestion
The `Callback` state can be used when you must execute an action in general to call an external service, and wait for an asynchronous response in the form of an event.

Once the response event is consumed, the workflow continues the execution, in general moving to the next state defined in the transition property.

Since the Callback state halts the execution until the event is consumed, you can configure an eventTimeout for it, and in case the event does not arrive in the configured time duration, the workflow continues the execution moving to the next state defined in the transition.
```